src/main/java/com/vmware/vim25/ws/CustomSSLTrustContextCreator.java

```diff
@@ -0,0 +1,41 @@
+package com.vmware.vim25.ws;
+
+import javax.net.ssl.*;
+import java.rmi.RemoteException;
+import java.security.KeyManagementException;
+import java.security.NoSuchAlgorithmException;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+public class CustomSSLTrustContextCreator {
+
+    private static AtomicBoolean contextAlreadyCreated = new AtomicBoolean(false);
+    private static SSLContext sslContext;
+
+    public static SSLContext getTrustContext(TrustManager trustManager) {
+        try {
+            if (contextAlreadyCreated.getAndSet(true)) {
+                return sslContext;
+            }
+
+            TrustManager[] trustManagers = new TrustManager[] { trustManager };
+            sslContext = SSLContext.getInstance("SSL");
+            sslContext.init(null, trustManagers, null);
+        }catch(NoSuchAlgorithmException e) {
+            throw new RemoteException("Unable to find suitable algorithm while attempting to communicate with remote server.", e);
+        } catch(KeyManagementException e) {
+            throw new RemoteException("An error occurred initializing SSL context due to a problem with key management.", e);
+        } finally {
+            return sslContext;
```

Comment:
Yes this is completely wrong, i should not have a return inside a finally block. I will correct this.